gitpod-io/gitpod
# [installer]: add HTTP_PROXY envvars to the Installer

## Description

Adds preliminary support for `HTTP_PROXY` (and `HTTPS_PROXY` and `NO_PROXY`) in the Installer. This adds an optional secret which receives the proxy settings from the KOTS CLI settings - the `NO_PROXY` envvar also hard-codes some additional parameters required for the application to work.

**Important** - this is not a full feature. It is merely the initial work to enable other teams to help out in their areas of expertise. This adds the appropriate envvars to the resources and wires them all up in a fashion that allows the Gitpod stack to run without error. However, it **DOES NOT** allow an image to be built or workspace to be run. There are additional tickets in #10769 that detail these tasks

An interesting thing of note is that the `@grpc/grpc-node` library does not respect wildcard `NO_PROXY` URLs. To address that, the `server` adds an addition `no_grpc_proxy` (which is the primary envvar for this library) and changes all resources it calls to the FQDN. See [the issue](https://github.com/grpc/grpc-node/issues/1293) for more details.

## Related Issue(s)

Fixes #12835 
Fixes #12820 
Fixes #12821

## How to test

> This will also need testing in a non-proxied instance to ensure no regressions are introduced

- Create a new server with [Squid](https://www.digitalocean.com/community/tutorials/how-to-set-up-squid-proxy-on-ubuntu-20-04) installed. This server must have unrestricted internet access
- Configure an airgapped installation and configure the k3s cluster to access the proxy
- Run `kubectl kots install gitpod --http-proxy <http-proxy> --https-proxy <https-proxy> --no-proxy <no-proxy>` to install Gitpod

### Configuring your cluster

A [repo](https://github.com/MrSimonEmms/gitpod-self-hosted-infrastructure) exists that will create a k3s cluster for you in Azure. Run `make azure-k3s` with the following settings in your `.auto.tfvars` file

```terraform
domain_name = "<your domain name>"
azure_k3s = true
enable_airgapped = true
http_proxy = "<your proxy settings - eg http://user:password@ip-address:3128">
```

## Release Notes

```release-note
[installer]: add HTTP_PROXY envvars to the Installer
```

## Documentation


## Werft options:
